entity-framework/core/providers/cosmos/planetary-docs-sample.md

```diff
@@ -0,0 +1,302 @@
+
+`PlanetaryDocs.Domain` hosts the domain classes, validation logic,
+and signature (interface) for data access.
+
+`PlanetaryDocs.DataAccess` contains the EF Core `DbContext`
+and an implementation of the data access service.
+
+- `DocsContext`
+  - Has model-building code that shows how to map ownership
+  - Uses value converters with JSON serialization to support primitives collection and nested
+complex types
+  - Demonstrates use of partition keys, including how to define them for the
+model and how to specify them in queries
+  - Provides an example of specifying the container by entity
+  - Shows how to turn off the discriminator
+  - Stores two entity types (aliases and tags) in the same container
+  - Uses a "shadow property" to track partition keys on aliases and tags
+  - Hooks into the `SavingChanges` event to automate the generation of audit snapshots
+- `DocumentService`
+  - Shows various strategies for C.R.U. operations
+  - Programmatically synchronizes related entities
+  - Demonstrates how to handle updates with concurrency to disconnected entities
+  - Uses the new `IDbContextFactory<T>` implementation to manage context instances
+
+`PlanetaryDocs` is a Blazor Server app.
+
+- Examples of JavaScript interop in the `TitleService`, `HistoryService`, and `MultiLineEditService`.
+- Uses keyboard handlers to allow keyboard-based navigation and input on the edit page
+- Shows a generic autocomplete component with debounce built-in
+- `HtmlPreview` uses a phantom `textarea` to render an HTML preview
+- `MarkDig` is used to transform markdown into HTML
+- The `MultiLineEdit` component shows a workaround using JavaScript interop for limitations with fields that have large input values
+- The `Editor` component supports concurrency. If you open a document twice in separate tabs and edit in both, the second will notify that changes were made and provide the option to reset or overwrite
+
+Your feedback is valuable! [File an issue](https://github.com/dotnet/EntityFramework.Docs/issues/new) to report defects or request changes (we also accept pull requests.)
+
+## Introducing Planetary Docs
+
+You may (or may not) know that Microsoft's [official documentation](https://docs.microsoft.com) runs entirely on open source. It uses markdown with some metadata enhancements to build the interactive documentation that .NET developers use daily. The hypothetical scenario for Planetary Docs is to provide a web-based tool for authoring the docs. It allows setting up the title, description, the alias of the author, assigning tags, editing markdown and previewing the HTML output.
+
+It's planetary because Azure Cosmos DB is "planetary scale." The app provides the ability to search documents. Documents are stored under aliases and tags for fast lookup, but full text search is available as well. The app automatically audits the documents (it takes snapshots of the document anytime it is edited and provides a view of the history).
```

Comment:
```suggestion
It's planetary because Azure Cosmos DB is "planetary scale". The app provides the ability to search documents. Documents are stored under aliases and tags for fast lookup, but full text search is available as well. The app automatically audits the documents (it takes snapshots of the document anytime it is edited and provides a view of the history).
```